utils/R/convert.input.R

```diff
@@ -1,333 +1,388 @@
-} # convert.input
-
-
-.get.file.deletion.commands <- function(files.to.delete) {
-  if(length(files.to.delete) > 0) {
-    tmp.dirs <- file.path(unique(dirname(files.to.delete)), 'tmp')
-    tmp.paths <- file.path(dirname(files.to.delete), 'tmp', basename(files.to.delete))
-    
-    tmp.dirs.string <- paste0("c(", paste(paste0("'", tmp.dirs, "'"), collapse=', '), ")")
-    tmp.path.string <- paste0("c(", paste(paste0("'", tmp.paths, "'"), collapse=', '), ")")
-    original.path.string <- paste0("c(", paste(paste0("'", files.to.delete, "'"), collapse=', '), ")")
-    
-    move.to.tmp <- paste0(
-      "dir.create(", tmp.dirs.string, ", recursive=TRUE, showWarnings=FALSE); ",
-      "file.rename(from=", original.path.string, ", to=", tmp.path.string, ")"
-    ) 
-    
-    replace.from.tmp <- paste0(
-      "file.rename(from=", tmp.path.string, ", to=", original.path.string, ");",
-      "unlink(", tmp.dirs.string, ", recursive=TRUE)"
-    )
-    
-    delete.tmp <- paste0(
-      "unlink(", tmp.dirs.string, ", recursive=TRUE)"
-    )
-    return(list(move.to.tmp=move.to.tmp, replace.from.tmp=replace.from.tmp, delete.tmp=delete.tmp))
-  } else {
-    return(NULL)
-  }
-} # .get.file.deletion.commands
+##' Convert input by applying fcn and insert new record into database
+##'
+##'
+##' @name convert.input
+##' @title convert.input
+##' @export
+##' @author Betsy Cowdery, Michael Dietze, Ankur Desai
+convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, start_date, 
+                          end_date, pkg, fcn, con = con, host, browndog, write = TRUE, 
+                          format.vars, overwrite = FALSE, exact.dates = FALSE, ...) {
+  input.args <- list(...)
+  
+  logger.debug(paste("Convert.Inputs", fcn, input.id, host$name, outfolder, formatname, 
+                     mimetype, site.id, start_date, end_date))
+  
+  n <- nchar(outfolder)
+  if (substr(outfolder, n, n) != "/") {
+    outfolder <- paste0(outfolder, "/")
+  }
+  
+  outname <- tail(unlist(strsplit(outfolder, "/")), n = 1)
+  
+  print(paste("start CHECK Convert.Inputs", fcn, input.id, host$name, outfolder, 
+              formatname, mimetype, site.id, start_date, end_date))
+  
+  existing.dbfile <- dbfile.input.check(siteid = site.id,
+                                        mimetype = mimetype, 
+                                        formatname = formatname, 
+                                        parentid = input.id, 
+                                        startdate = start_date,
+                                        enddate = end_date, 
+                                        con = con, 
+                                        hostname = host$name, 
+                                        ignore.dates = TRUE)
+  
+  print(existing.dbfile, digits = 10)
+  print("end CHECK")
+  
+  if (nrow(existing.dbfile) > 0) {
+    
+    existing.input <- db.query(paste0("SELECT * FROM inputs WHERE id=", existing.dbfile[["container_id"]]),con)

```

Comment:
existing.dbfile can never be >1 because the function that creates it, dbfile.input.check, selects the last one that was made and cuts out the other possible ones.